modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerIndexForceRebuildTest.java

```diff
@@ -0,0 +1,432 @@
+
+    /** */
+    private static final String CACHE_NAME_NO_GRP = "cache_no_group";
+
+    /** */
+    private static final String CACHE_NAME_NON_EXISTING = "non_existing_cache";
+
+    /** */
+    private static final String GRP_NAME_1 = "group_1";
+
+    /** */
+    private static final String GRP_NAME_2 = "group_2";
+
+    /** */
+    private static final String GRP_NAME_NON_EXISTING = "non_existing_group";
+
+    /** */
+    private static final int GRIDS_NUM = 3;
+
+    /** */
+    private static final int LAST_NODE_NUM = GRIDS_NUM - 1;
+
+    /**
+     * Set containing names of caches for which index rebuild should be blocked.
+     * See {@link BlockingIndexing}.
+     */
+    private static Set<String> cacheNamesBlockedIdxRebuild = new GridConcurrentHashSet<>();
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        ListeningTestLogger testLog = new ListeningTestLogger(log);
+
+        return super.getConfiguration(igniteInstanceName)
+            .setGridLogger(testLog);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        cleanPersistenceDir();
+
+        startupTestCluster();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+
+        super.afterTestsStopped();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        cacheNamesBlockedIdxRebuild.clear();
+    }
+
+    /** */
+    private void startupTestCluster() throws Exception {
+        for (int i = 0; i < GRIDS_NUM; i++ ) {
+            GridQueryProcessor.idxCls = BlockingIndexing.class;
+            startGrid(i);
+        }
+
+        IgniteEx ignite = grid(0);
+
+        ignite.cluster().active(true);
+
+        createAndFillCache(ignite, CACHE_NAME_1_1, GRP_NAME_1);
+        createAndFillCache(ignite, CACHE_NAME_1_2, GRP_NAME_1);
+        createAndFillCache(ignite, CACHE_NAME_2_1, GRP_NAME_2);
+
+        createAndFillThreeFieldsEntryCache(ignite, CACHE_NAME_NO_GRP, null, Collections.singletonList(complexIndexEntity()));
+    }
+
+    /**
+     * Checks error messages when trying to rebuild indexes for
+     * non-existent cache of group.
+     */
+    @Test
+    public void testEmptyResult() {
+        injectTestSystemOut();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "indexes_force_rebuild",
+            "--node-id", grid(LAST_NODE_NUM).localNode().id().toString(),
+            "--cache-names", CACHE_NAME_NON_EXISTING));
+
+        String cacheNamesOutputStr = testOut.toString();
+
+        assertTrue(cacheNamesOutputStr.contains("WARNING: Indexes rebuild was not started for any cache. Check command input."));
+
+        testOut.reset();
```

Comment:
It's strange that you have both log listeners and log String. Why?

Reply:
Loglistener is used to check messages in node's log. testOut is used to check messages in command output.